src/main.cpp

```diff
@@ -1,18 +1,77 @@
 #include <pros/vision.hpp>
 #include <pros/rtos.hpp>
+#include <cmath>
+#include <optional>
+#include <iostream>
 #include "main.hpp"
 
 void initialize() {}
 void disabled() {}
 void competition_initialize() {}
 void autonomous() {}
 
-#define VISION_PORT 1
+namespace ports {
+	constexpr int VISION = 1;
+}
+
+namespace measures {
+	constexpr int GOAL_HEIGHT = 64; // 25 in to cm
+	constexpr int SENSOR_HEIGHT = 44; // 17.5 in to cm
+	constexpr float VERTICAL_FOV = 0.715585; // 41 deg to radians
+	constexpr float LENS_ANGLE = 0.261799; // 15 deg to radians
+}
+
+constexpr int SIG_ERR = 255;
+
+class Sensor {
+private:
+	pros::Vision m_sensor;
+	pros::vision_signature_s_t m_sig;
+public:
+	Sensor(int port, pros::vision_signature_s_t sig): m_sensor(port), m_sig(sig) {
```

Comment:
space before colon (I think this is covered by clang format)
use brace init for m_sensor since it has a constructor

Reply:
also both params should be const

Reply:
`Sensor(const int port, const pros::vision_signature_s_t sig) : m_sensor(port), m_sig(sig) { ... }`

Reply:
no, use east const

Reply:
Okay so for east const is the ampersand on the const or on the variable name

Reply:
```
int const value
int const& ref
int const* ptr
int const* const cptr
```